app/models/targeting.rb

```diff
@@ -0,0 +1,41 @@
+class Targeting < ActiveRecord::Base
+
+  STATIC_TYPE = 'static_query'
+  DYNAMIC_TYPE = 'dynamic_bookmark'
+  TYPES = {STATIC_TYPE => N_('Static Query'), DYNAMIC_TYPE => N_('Dynamic Bookmark')}
+
+  belongs_to :user
+  belongs_to :bookmark
+
+  has_many :targeting_hosts, :dependent => :destroy
+  has_many :hosts, :through => :targeting_hosts
+  has_many :template_invocations, :dependent => :destroy
+
+  validates :targeting_type, :presence => true, :inclusion => Targeting::TYPES.keys
+  validates :bookmark, :presence => true, :if => "dynamic?"
+
+  before_create :assign_search_query, :if => :static?
```

Comment:
I tend to prefer to display a warning at this time, the main reason is that
if someone schedule something for the future and wants to change it, it
becomes a simple change vs changing all pre-generated targets in bulk (and
heavy updates in the db).

maybe we can add a state, like planning or scheduled, vs locked/deployed
where you can edit only while you are in the planning phase and not in the
locked/executed phase?